### PR TITLE
ref(ts): Convert u2f components to typescript

### PR DESCRIPTION
**Pull request by k-fish**
This converts the remaining u2f components to typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx [hunk 1]
```diff
@@ -47,7 +47,13 @@ class U2fContainer extends React.Component<Props, State> {
       <div className={className}>
         {authenticators.map(({id, ...other}) => {
           if (id === 'u2f' && other.challenge) {
-            return <U2fSign key={id} {...this.props} challengeData={other.challenge} />;
+            return (
+              <U2fSign
+                key={id}
+                {...this.props}
+                challengeData={other.challenge as U2fSign['props']['challengeData']}
```
Comment (k-fish) [inline src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx:54]: Not sure about this `as` here, could move the challenge data type into `app/types` so authenticator can use it, but then types would to import `u2f-api`. Thoughts?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx:54]: It should be fine if it's just importing the type, yeah?

File: src/sentry/static/sentry/app/components/u2f/u2finterface.tsx [hunk 2]
```diff
@@ -6,7 +6,29 @@ import * as Sentry from '@sentry/react';
...
+  authenticateRequests: u2f.SignRequest;
+  registerRequests: u2f.RegisterRequest;
+};
+
+type Props = {
+  challengeData: ChallengeData;
+  flowMode: string;
+  style?: React.CSSProperties;
+  silentIfUnsupported: boolean;
+  onTap: ({response, challenge}) => Promise<any>;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/u2f/u2finterface.tsx:19]: Is there a better return type than `any` here?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx [hunk 3]
```diff
@@ -8,8 +8,8 @@ import U2fSign from './u2fsign';
 
 type Props = {
   api: Client;
-  displayMode?: string;
-  onTap?: ({response, challenge}: {response: string; challenge: string}) => void;
+  displayMode?: U2fSign['props']['displayMode'];
+  onTap: U2fSign['props']['onTap'];
   className?: string;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx:13]:
```suggestion
  onTap: U2fSign['props']['onTap'];
  displayMode?: U2fSign['props']['displayMode'];
  className?: string;
```

File: src/sentry/static/sentry/app/components/u2f/u2finterface.tsx [hunk 4]
```diff
@@ -6,7 +6,29 @@ import * as Sentry from '@sentry/react';
...
+};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/u2f/u2finterface.tsx:20]:
```suggestion
type Props = {
  challengeData: ChallengeData;
  flowMode: string;
  silentIfUnsupported: boolean;
  onTap: ({response, challenge}) => Promise<any>;
   style?: React.CSSProperties;
};
```

File: src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx (hunk 1 above)
Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/u2f/u2fContainer.tsx:54]: I would move the challenge data to types and Imo it's fine to import the `u2f-api `

File: src/sentry/static/sentry/app/components/u2f/u2finterface.tsx (hunk 2 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/components/u2f/u2finterface.tsx:18]: I didn't see any returns in the code, so undefined I guess.

File: src/sentry/static/sentry/app/components/u2f/u2finterface.tsx [hunk 5]
```diff
@@ -188,7 +208,7 @@ class U2fInterface extends React.Component {
                   support,
                 }
               ),
-            }[deviceFailure]
+            }[deviceFailure || '']
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/u2f/u2finterface.tsx:211]: you don't need `' '` here 😉
```suggestion
            }[deviceFailure]
```

Comment (k-fish) [reply src/sentry/static/sentry/app/components/u2f/u2finterface.tsx:211]: It's `string | null`, null can't be used as an index type.

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/u2f/u2finterface.tsx:211]: oh yeah..typescript :)

**Review by priscilawebdev — APPROVED** [review]

**Pull request merged**